mqtt/mqtt-broker/src/snapshot.rs

```diff
@@ -39,28 +40,33 @@ impl SessionSnapshot {
         client_info: ClientInfo,
         subscriptions: HashMap<String, Subscription>,
         waiting_to_be_sent: VecDeque<Publication>,
+        waiting_to_be_acked: VecDeque<Publish>,
         last_active: DateTime<Utc>,
     ) -> Self {
         Self {
             client_info,
             subscriptions,
             waiting_to_be_sent,
+            waiting_to_be_acked,
             last_active,
         }
     }
 
+    #[allow(clippy::type_complexity)]
```

Comment:
well, I agree with clippy. just as an idea: can we get a mutable reference instead and make mem::take or mem replace for each field instead?

Reply:
Yeah, it's not super pretty, and I was thinking about that as well. The main problem here is that we're trying to deconstruct it from other modules that don't have access to the private fields. 

I've tried to see how it's done in other places. I believe `pub(crate)` on fields is one approach, and having sub-struct `Parts` with pub fields is another one. 

Taking fields one by one with mem::tak is the new one that you're suggesting. But, IMO, it is even less prettier.